src/main/groovy/io/sentry/android/gradle/GenerateSentryProguardUuidTask.java

```diff
@@ -0,0 +1,54 @@
+package io.sentry.android.gradle;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.file.DirectoryProperty;
+import org.gradle.api.file.RegularFile;
+import org.gradle.api.provider.Property;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.tasks.Internal;
+import org.gradle.api.tasks.OutputDirectory;
+import org.gradle.api.tasks.TaskAction;
+
+import java.io.BufferedWriter;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.util.UUID;
+
+abstract class GenerateSentryProguardUuidTask extends DefaultTask {
```

Comment:
I wrote this in java in preparation for a Kotlin rewrite which feels like a well spent effort.